compiler/rustc_interface/src/passes.rs

```diff
@@ -1078,16 +1078,22 @@ pub(crate) fn start_codegen<'tcx>(
     codegen_backend: &dyn CodegenBackend,
     tcx: TyCtxt<'tcx>,
 ) -> Box<dyn Any> {
+    // Hook for UI tests.
+    check_for_rustc_errors_attr(tcx);
+
+    // Don't run this test assertions when not doing codegen. Compiletest tries to build
```

Comment:
This comment feels misplaced 🤔

Reply:
Or at least I'm not seeing its *point*. Are the ui tests relying on `//@ build-fail`?